docs/adr/0002_persisted.md

```diff
@@ -0,0 +1,53 @@
+# Introduce `PersistedWallet`
+
+* Status: accepted
+* Deciders: [list everyone involved in the decision]
+* Date: 2024-08-19
+
+Technical Story:
+
+* [ADR-0001](./0001_persist.md)
+* [#1514 - refactor(wallet)!: rework persistence, changeset, and construction](https://github.com/bitcoindevkit/bdk/pull/1514)
+* [#1547 - Simplify wallet persistence](https://github.com/bitcoindevkit/bdk/pull/1547)
+
+## Context and Problem
+
+BDK v1.0.0-beta.1 introduced new persistence traits `PersistWith<Db>` and `PersistAsyncWith<Db>`. The design was [slightly overwrought][0], the latter proving difficult to implement by wallet users.
+
+## Decision Drivers
+
+We would like persistence operations to be both safe and ergonomic. It would cumbersome and error prone if users were expected to ask the wallet for a changeset at the correct time and persist it on their own. Considering that the wallet no longer drives its own database, an ideal interface is to have a method on `Wallet` such as [`persist`][1] that the user will call and everything will "just work."
+
+## Chosen option
+
+Introduce a new type `PersistedWallet` that wraps a BDK `Wallet` and provides a convenient interface for executing persistence operations aided by a `WalletPersister`.
+
+`PersistedWallet` internally calls the the methods of the `WalletPersister` trait. We do this to ensure consistency of the create and load steps particularly when it comes to handling staged changes and to reduce the surface area for footguns.
+
+The two-trait design was kept in order to accommodate both blocking and async use cases. For `AsyncWalletPersister` the definition is modified to return a [`FutureResult`][2] which is a rust-idiomatic way of creating something that can be polled by an async runtime. For example in the case of `persist` the implementer writes an `async fn` that does the persistence operation and then calls `Box::pin` on that.
+```rust
+impl AsyncWalletPersister for MyCustomDb {
+    ...
+
+    fn persist<'a>(persister: &'a mut Self, changeset: &'a ChangeSet) -> FutureResult<'a, (), Self::Error>
+    where
+        Self: 'a,
+    {
+        let persist_fn = async move |changeset| {
+            // perform write operation...
+            Ok(())
+        };
+    
+        Box::pin(persist_fn)
+    }
+}
+```
```

Comment:
Maybe add links referencing the current implementations for `WalletPersister` too? :think: